perception/autoware_traffic_light_map_based_detector/src/traffic_light_map_based_detector_node.cpp

```diff
@@ -303,47 +221,41 @@ bool MapBasedDetector::getTrafficLightRoi(
 
   // for roi.x_offset and roi.y_offset
   {
-    tf2::Vector3 map2tl = getTrafficLightTopLeft(traffic_light);
-    tf2::Vector3 camera2tl = tf_map2camera.inverse() * map2tl;
+    tf2::Vector3 tl_map = traffic_light_utils::getTrafficLightTopLeft(traffic_light);
+    tf2::Vector3 tl_camera_optical = tf_map2camera.inverse() * tl_map;
     // max vibration
-    const double max_vibration_x =
-      std::sin(config.max_vibration_yaw * 0.5) * camera2tl.z() + config.max_vibration_width * 0.5;
-    const double max_vibration_y = std::sin(config.max_vibration_pitch * 0.5) * camera2tl.z() +
-                                   config.max_vibration_height * 0.5;
-    const double max_vibration_z = config.max_vibration_depth * 0.5;
+    tf2::Vector3 point3d = utils::getVibrationMarginTopLeft(
+      tl_camera_optical, config.max_vibration_pitch, config.max_vibration_yaw,
+      config.max_vibration_height, config.max_vibration_width, config.max_vibration_depth);
```

Comment:
1. `getVibrationMarginTopLeft` and `getVibrationMarginBottomRight` are almost same. integrate it into one `getVibrationMargin`
2. hand-over all the config margins are complicate. give a variable as `Config` type. to do so, may need to separate the type definition

```suggestion
    tf2::Vector3 margin = utils::getVibrationMarginTopLeft(tl_camera_optical.z, config);
    tf2::Vector3 point3d = tl_camera_optical - margin;
```

Reply:
Regarding 1
Fixed in https://github.com/autowarefoundation/autoware_universe/pull/10353/commits/ecc788bc953dae001a36d6a5c8b02a24c2345f28

Regarding 2
I agree with you. But I leave it at that.
`Config` is defined in private member of `TrafficLightMapBasedDetectorNode`. So I would not like to pass as argument if use function like `utils`. And this `Config` is not sophisticated so I have the plann which I will reduce variable in it.